common/build.rs

```diff
@@ -91,42 +99,81 @@ fn gen_binding<P: AsRef<Path>>(
 
 fn rust_swig_expand(from: &Path, out_dir: &Path) {
     println!("Run rust_swig_expand");
-    let swig_gen = rust_swig::Generator::new(LanguageConfig::JavaConfig(JavaConfig::new(
-        get_java_codegen_output_directory(&out_dir),
-        "com.ironcorelabs.sdk".into(),
-    )))
-    .merge_type_map("chrono_support", include_str!("chrono-include.rs"))
-    .rustfmt_bindings(true)
-    .remove_not_generated_files_from_output_directory(true); //remove outdated *.java files
-    swig_gen.expand("rust_swig_test_jni", from, out_dir.join("lib.rs"));
+
+    println!("{}", include_str!("chrono-include.rs"));
+
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "cpp")]{
+            let name = "ironoxide_cpp";
+            let config = CppConfig::new(get_cpp_codegen_output_directory(), "sdk".into())
+              .cpp_optional(CppOptional::Boost)
+              .separate_impl_headers(true);
+
+            let swig_gen = rust_swig::Generator::new(LanguageConfig::CppConfig(config))
+              .merge_type_map("chrono_support", include_str!("../cpp/chrono-include.rs"));
+        } else{
+            let name = "ironoxide_jvm";
+            let swig_gen = rust_swig::Generator::new(LanguageConfig::JavaConfig(JavaConfig::new(
+                get_java_codegen_output_directory(&out_dir),
+                "com.ironcorelabs.sdk".into(),
+            )))
+            .merge_type_map("chrono_support", include_str!("chrono-include.rs"));
+        }
+    }
+
+    swig_gen
+        .rustfmt_bindings(true)
+        .remove_not_generated_files_from_output_directory(true) //remove outdated *.java or cpp files
+        .expand(name, from, out_dir.join("lib.rs"));
 }
 
-fn get_java_codegen_output_directory(out_dir: &Path) -> PathBuf {
-    let path = out_dir
-        .join("java")
-        .join("com")
-        .join("ironcorelabs")
-        .join("sdk");
-    if !path.exists() {
-        std::fs::create_dir_all(&path).expect(
-            "Couldn't create codegen output directory at OUT_DIR/java/com/ironcorelabs/sdk.",
-        );
+cfg_if::cfg_if! {
+    if #[cfg(feature = "cpp")]{
+        fn get_cpp_codegen_output_directory() -> PathBuf {
+            //let out_dir = env::var("OUT_DIR").expect("no OUT_DIR, but cargo should provide it");
+            let path = Path::new("generated").join("sdk");
+            if !path.exists() {
+                std::fs::create_dir_all(&path).expect(format!("Couldn't create codegen output directory at {:?}.", path).as_str());
```

Comment:
does this get removed by cargo clean? if not, should it?

Reply:
It does not, because it's not under target. I guess we should decide if we want the headers to be under target or not.